File: XeniaAndRingroad.java
import java.util.Scanner;

public class XeniaAndRingroad {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        
        int current = 1;
        long time = 0;
        
        for (int i = 0; i < m; i++) {
            int target = scanner.nextInt();
            if (target >= current) {
                time += target - current;
            } else {
                time += n - current + target;
            }
            current = target;
        }
        
        System.out.println(time);
        
        scanner.close();
    }
}

File: XeniaAndRingroad.cpp
#include <iostream>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    
    int current = 1;
    long long time = 0;
    
    for (int i = 0; i < m; i++) {
        int target;
        cin >> target;
        if (target >= current) {
            time += target - current;
        } else {
            time += n - current + target;
        }
        current = target;
    }
    
    cout << time << endl;
    
    return 0;
}